tenant/repository.go

```diff
@@ -162,35 +193,69 @@ func (s DBService) DeleteTenant(tenantID uuid.UUID) error {
 	return s.db.Unscoped().Delete(&Tenant{ID: tenantID}).Error
 }
 
-type NilService struct {
+func (s DBService) NewTenantRepository(tenantID uuid.UUID) Repository {
+	return DBTenantRepository{service: s, tenantID: tenantID}
```

Comment:
why does the repository need to handle a reference to the service?

Reply:
Because the repository is just a wrapper (that keeps the `tenantID`) and calls the service. 
To be honest, in the first stages of my work I also totally changed this representation of the access to the DB. But as part of my attempt to keep the PR as small as possible, I didn't introduce this change here. I just created the wrapper so I don't have to pass the `tenantID` as a parameter everywhere, so I ended up with the approach of having the service as the source of all logic and using the repo just as the wrapper. 
However, during the time the PR went bigger and bigger and I should have changed this repo/service solution as well... I'll try to move the relevant logic from the service to the repository and I'll change also the places where it is used - or, at least, I'll try my best ;-)

Reply:
ok, I tried my best and this is the result: 1c4a496 I hope that it looks better now

Reply:
thanks @MatousJobanek this is much better indeed! 🙌